### ISSUE TITLE
Bug in the assignment of NBFIXes in Amber topologies

### ISSUE THREAD

**Issue by milafternoon** [ParmEd/ParmEd#1214]
Looking further into the Amber vs Gromacs treatment of NBFIXes, I started comparing Gromacs topologies converted to Amber and found that (again) their energies don't match exactly. I could trace back the issue to some discrepancies in how LENNARD_JONES_ACOEF and LENNARD_JONES_BCOEF are updated - all NBFIXed terms are correct, but then these modified parameters pop up in several other entries that should not be changed from the default L-B rules.

After some digging, I noticed that  `_set_nonbonded_tables()` in `_amberparm.py` introduces this bug in the line:

`i, j = min(i, j-1), max(i, j-1)`

What happens is, sometimes `i` from the outer loop gets assigned to `j`, and `j` gets reassigned in the inner loop - so that now we assign the NBFIXed parameters to a semi-random pair of indices. A simple solution would be just to use a different variable name:

`m, n = min(i, j-1), max(i, j-1)`
and then 
`idx = self.parm_data['NONBONDED_PARM_INDEX'][ntypes*m+n] - 1`

Best regards,
Miłosz

**Comment by swails** [comment]
Sorry for missing this.  Thank you for the report.  If you implement this change does it resolve the discrepancies?

**Comment by milafternoon** [comment]
Yes, in my case that was what fixed the discrepancy in energy decomposition. I also did a comparison of expected vs actual Amber COEFs using params from the Gromacs file, and they were indeed identical after changing this.

**Comment by milafternoon** [comment]
Hi, was there any progress on this and [the other report](https://github.com/ParmEd/ParmEd/issues/1213)? I still have to manually modify these two chunks of code every time I install parmed on a new machine, and I know it's a matter of time before I forget to do it somewhere.
